### PR TITLE
Add watchdogs to environment struct

### PR DESCRIPTION
**Pull request by erlingrj**
These changes move the watchdogs to the environment struct and thus make the watchdog compatible with the future enclave support. The corresponding changes in reactor-c also address the termination issue for the watchdogs as described in #2150 

See the corresponding reactor-c PR : https://github.com/lf-lang/reactor-c/pull/341

### CODE REVIEW

File: core/src/main/java/org/lflang/generator/c/CWatchdogGenerator.java [hunk 1]
```diff
@@ -83,8 +92,7 @@ protected static int generateInitializeWatchdogs(CodeBuilder code, ReactorInstan
     if (foundOne) {
       code.pr(temp.toString());
     }
-    code.pr("SUPPRESS_UNUSED_WARNING(_lf_watchdog_count);");
-    return watchdogCount;
+    enclaveInfo.numWatchdogs += watchdogCount;
```
Comment (edwardalee) [inline core/src/main/java/org/lflang/generator/c/CWatchdogGenerator.java:95]: This doesn't look quite right.  Won't `watchdogCount` always be 0?

**Review by edwardalee — CHANGES_REQUESTED** [review]
Looks good to me except one possible flaw identified. Is it a flaw?

File: core/src/main/java/org/lflang/generator/c/CWatchdogGenerator.java (hunk 1 above)
Comment (erlingrj) [reply core/src/main/java/org/lflang/generator/c/CWatchdogGenerator.java:97]: I think you overlooked the problem the place where it is incremented. Right above

**Review by edwardalee — APPROVED** [review]
LGTM!  The LSP test failure is probably unrelated (these tests are randomized) and probably due to the code generator generating blank lines, which the Python validator in the LSP doesn't like but the Python interpreter apparently has no problem with.

**Pull request merged**